src/App.tsx

```diff
@@ -15,48 +16,108 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  sortByAlph = 'alphabetically',
+  sortByLength = 'length',
+  byDefault = '',
+}
+
+function getPreparedGoods(
+  goods: string[],
+  sortButton: SortType,
+  isReversed: boolean,
+): string[] {
+  const preparedGoods = [...goods];
+
+  if (sortButton) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortButton) {
+        case SortType.sortByAlph:
+          return good1.localeCompare(good2);
+
+        case SortType.sortByLength:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortButton, setSortButton] = useState(SortType.byDefault);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const sortedGoods = getPreparedGoods(goodsFromServer, sortButton, isReversed);
+
+  const getReversedGoods = () => (
+    !isReversed
+      ? setIsReversed(true)
+      : setIsReversed(false)
+  );
```

Comment:
Use a more descriptive name for the variable, `toggleReversed` for example. And use functional setState to get the value based on the previous one.